### PR TITLE
Fix: Handle undefined language field in multilingual content (#7309)

### LINKED ISSUE #7309: Content without categorization behavior causes errors in view on multilingual sites
**Describe the bug**
Error occurred:

```javascript
Cannot read properties of undefined (reading 'token')
```

**To Reproduce**

1. Create a multilingual website.
2. Create a content type without the Categorization behavior.
3. Create an object of that type and access the view.

**Expected behavior**
It should show the content type view without error.

**Software (please complete the following information):**

- Volto Version 18.25.0
- Plone Version 6.1.2

**Additional context**
This error started occurring after: https://github.com/plone/volto/pull/7125
I believe the line with the problem is: https://github.com/plone/volto/blob/4ef468c57298038dd0df623b6abef4efdbaf7b82/packages/volto/src/components/theme/AlternateHrefLangs/AlternateHrefLangs.jsx#L11

### PR DESCRIPTION
**Pull request by ravi-aman**
# Fix language token crash in multilingual content

## The Problem

I ran into this bug while working with custom content types. When you add translation support (`plone.translatable` behavior) to content but don't include the full Dublin Core behavior (which has the Language field), Volto crashes with:

```
Cannot read properties of undefined (reading 'token')
```

This happens because the code assumes `content.language` is always an object with a `.token` property, but it can actually be:
- `undefined` (no language set)
- A plain string like `"en"` 
- An object with `.token` like `{ token: "en", title: "English" }`

## What I Changed

Added a helper function `getLanguageToken()` that safely handles all three cases. Then updated these components to use it:

- **AlternateHrefLangs** - The one throwing the original error
- **ManageTranslations** - 7 places where it accessed `content.language.token`
- **Edit component** - 3 places in the Helmet lang attribute and menu items

## Testing

I added 8 unit tests covering all the edge cases:
- undefined/null values
- string values  
- object with token
- object without token
- other invalid types

The fix is backwards compatible - existing code with proper language objects keeps working, and now content without Dublin Core doesn't crash anymore.

## Files Changed

- `src/helpers/Content/Content.js` - Added `getLanguageToken()` helper
- `src/helpers/Content/Content.test.js` - Added test coverage
- `src/components/theme/AlternateHrefLangs/AlternateHrefLangs.jsx` - Use helper
- `src/components/manage/Multilingual/ManageTranslations.jsx` - Use helper (7 spots)
- `src/components/manage/Edit/Edit.jsx` - Use helper (3 spots)
- `news/7309.bugfix` - Changelog entry

Closes #7309

### CODE REVIEW

File: packages/volto/src/helpers/Content/Content.js [hunk 1]
```diff
@@ -108,3 +108,23 @@ export function flattenStaticBehaviors(result) {
...
+ * @param {Object|string|undefined} language - Language field from content
+ * @returns {string|null} Language token (e.g., 'en', 'de') or null if not available
+ * @example
+ * getLanguageToken(undefined) // returns null
+ * getLanguageToken('en') // returns 'en'
+ * getLanguageToken({ token: 'en', title: 'English' }) // returns 'en'
+ */
+export function getLanguageToken(language) {
+  if (!language) return null;
+  if (typeof language === 'string') return language;
```
Comment (wesleybl) [inline packages/volto/src/helpers/Content/Content.js:127]: The `language` attribute in Plone is of type `schema.Choice`. Therefore, it will always be an object. I suggest a simpler fix, with:

```javascript
content?.language.token
```

**Review by wesleybl — CHANGES_REQUESTED** [review]

**Comment by erral** [comment]
Backport also to 18.x.x please

**Comment by wesleybl** [comment]
Closed in favor of #8100.
